dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -80,21 +80,21 @@
     Invalid file format: {{ file_format }}
     Snapshot functionality requires file_format be set to 'delta'
   {%- endset %}
+
+  {% set target_relation_exists, target_relation = get_or_create_relation(
+          database=none,
+          schema=model.schema,
+          identifier=target_table,
+          type='table') -%}
   
-  {%- if file_format != 'delta' -%}
+  {%- if not target_relation_exists.is_delta -%}
```

Comment:
I didn't initially appreciate the difference between `is_delta` being a property of the `Relation`, whereas `config.file_format` is an attribute of the model `node`.

If this is the first time we're running the snapshot, the relation won't yet exist, so we'll want to check the snapshot config to ensure the user has set the file format to be delta. If this is a subsequent run, then it is right and proper to check the existing table and confirm its type to be delta.

So I think that could look something like:
```suggestion
{%- if not (target_relation.is_delta or (
    not target_relation_exists and
    config.get('file_format', validator=validation.any[basestring]) == 'delta'
)) -%}
```

Reply:
@jtcohen6 I've cleaned up the error messages a bit, let me know what you think of it